### DISCUSSION TITLE
A little bit confused about using sessions globally!

### DISCUSSION THREAD

**Discussion by mikatpt** [nextauthjs/next-auth#985] [Help]
I've been using the _app level Provider to pass sessions down to my components, and had some questions-
Within getServerSideProps, I took a look at the passed in props and saw that **session** is attached to the pageProps, such that I can do this in my pages: 
```
export async function getServerSideProps({ session }) {
  return { props: { session } };
}
```

Is this an acceptable way to pass sessions down, or should I still use the below await pattern?

```
export async function getServerSideProps({ req }) {
  const session = await getSession({ req });
  return { props: { session } };
}
```

I realize that getSession() is a hook, so I _think_ it's not that big of a deal to call it repeatedly, but I can't tell if it's redundant with the global Provider there already.

Forgive me if this is a somewhat simple question - I'm rather new to a lot of this and the material is just foreign enough that I can't tell if I'm doing things wrong or not!

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by TimNZ** [comment]
The object passed to getServerSideProps will never have session prop, 
use the 2nd code example.
getServerSideProps is called before the _app() component is called.

Where did you see an example of the first code block usage?

getSession isn't a hook, useSession is a hook.

https://github.com/nextauthjs/next-auth/issues/345

**Reply by mikatpt** [reply]
Okay that does clear up my misunderstanding. I thought that _app gets called first, so when I put a console.log(pageProps) in it and saw **session** attached, I assumed I was missing something with how the context Provider works. I'm not sure why it still worked when I was testing, but I think I might have just done something funky while I was tinkering with it.

Thank you!
